service_container.rst

```diff
@@ -1072,6 +1072,44 @@ If you want to pass the second, you'll need to :ref:`manually wire the service <
     and the automatically loaded service will be passed - by default - when you type-hint
     ``SiteUpdateManager``. That's why creating the alias is a good idea.
 
+Injecting the content of YAML file as argument
+----------------------------------------------
+
+You can automatically parse a YAML file and injecting it as an argument by referencing the **absolute path** to the file,
+or the relative path from **your configuration file**:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            # ...
+
+            App\ServiceWithYamlInside:
```

Comment:
What about `App\ServiceWithArrayConfigFromYamlFile` or something more like it?
The current naming let think that we may get a YAML string.

Reply:
Agreed!